### PR TITLE
Add an option to let rustfmt 0.7 work

### LINKED ISSUE #212: :RustFmt not work with rustfmt 0.7.0
as the topic

### PR DESCRIPTION
**Pull request by Marwes**
Added an option to let this plugin work with both rustfmt < 0.6 and >=
0.7.

Fixes #212

### CODE REVIEW

**Comment by jonhoo** [comment]
ping @steveklabnik @chris-morgan auto-formatting in vim is currently impossible for nightly users because of this. Setting `--emit=files` in `g:rustfmt_options` doesn't work because `rustfmt` still complains about an unrecognized argument `write-mode`.

**Comment by jonhoo** [comment]
@Marwes I wonder if this PR should actually run `rustfmt --version` and dynamically switch what flags to use because of that. Setting a global flag seems pretty unfortunate in the face of `rustup` and per-directory overrides.

(for those running into this issue, I've worked around this on top of this PR by setting my `rustfmt_command` to `rustfmt +nightly`, and that works pretty well)

**Comment by jalexvig** [comment]
Any progress on this? Still not working with rustfmt 0.8.2

**Comment by jalexvig** [comment]
@jonhoo Is the best way to do this using cargo?

`cargo install --list | sed -n "s/rustfmt-nightly v\(.*\):/\1/p"`

**Comment by jonhoo** [comment]
@jalexvig I'm not entirely sure what you mean? I just applied the diff from this PR directly to my `rust.vim` checkout (in `~/.local/share/nvim/plugged/rust.vim` if you're using `Plug` in `neovim`), and then set
```vim
let g:rustfmt_command = "rustfmt +nightly"
let g:rustfmt_emit_files = 1
```

**Comment by jalexvig** [comment]
@jonhoo whoops I was talking about getting the `rustfmt` version but totally missed `rustfmt --version` as you pointed out originally. Will delete that comment!

**Comment by jalexvig** [comment]
Okay here's the dynamic way of doing this... @Marwes will you update the PR?

`let g:rustfmt_emit_files = 1 - (system("rustfmt --version") =~ "rustfmt 0.[0-6].\.*")`

**Comment by korken89** [comment]
I tested the auto detect and it work as expected, I made a branch with the addition here: https://github.com/korken89/rust.vim/tree/rustfmt_0.7
If @Marwes does not respond soon, should I make a PR to get this in?

**Comment by Marwes** [comment]
Sorry for forgetting about this, I cherry-picked the auto detection from @korken89 and updated the docs for the setting.

Unsure which to ping for this repo so cc @steveklabnik (last one to merge a PR)

**Comment by korken89** [comment]
Awesome, thanks for adding! Lets hope they merge now :)

**Comment by OddBloke** [comment]
This is working for me (with nightly Rust); I'd love to see it merged!

**Comment by jalexvig** [comment]
@rust-highfive any news on this?

**Comment by Mark-Simulacrum** [comment]
@steveklabnik @rust-lang/dev-tools Can we find a reviewer for this pull request?

**Comment by nrc** [comment]
r+ from the rustfmt point of view, I have no idea about .vim files though.

**Comment by yoshuawuyts** [comment]
Tested locally, can confirm this patch works :tada:

File: autoload/rustfmt.vim [hunk 1]
```diff
@@ -19,15 +19,29 @@ if !exists("g:rustfmt_fail_silently")
 	let g:rustfmt_fail_silently = 0
 endif
 
+if !exists("g:rustfmt_emit_files")
+	let g:rustfmt_emit_files = 1 - (system("rustfmt --version") =~ "rustfmt 0.[0-6].\.*")
```
Comment (jonhoo) [inline autoload/rustfmt.vim]: Hmm, shouldn't this use `g:rustfmt_command`?

**Pull request merged**